runtime/System/Console/Hawk/Runtime.hs

```diff
@@ -20,18 +21,28 @@ import System.Console.Hawk.Runtime.Base
 
 data SomeRows = forall a. Rows a => SomeRows a
 
+-- Separate the lazy and strict versions so we don't get blocked
```

Comment:
I would prefer to use the "write to a temporary and copy over original on success" approach and to remain lazy.